File: Frontend/AppFavoresMovil/app/src/main/java/com/example/appfavoresmovil/data/modelos/favores/FavorCreacion.kt
package com.example.appfavoresmovil.data.modelos.favores

import com.google.gson.annotations.SerializedName

data class FavorCreacion(
    @SerializedName("titulo") val titulo: String,
    @SerializedName("descripcion") val descripcion: String,
    @SerializedName("modalidad") val modalidad: String,
    @SerializedName("ubicacion") val ubicacion: String?,
    @SerializedName("tipoCompensacion") val tipoCompensacion: String,
    @SerializedName("cantidadDinero") val cantidadDinero: Float?,
    @SerializedName("recompensaFavor") val recompensaFavor: String?
)

File: Frontend/AppFavoresEscritorio/AppFavoresEscritorio/Modelos/Favores/FavorCreacion.cs
using System.Text.Json.Serialization;

namespace AppFavoresEscritorio.Modelos.Favores;

public class FavorCreacion
{
    [JsonPropertyName("titulo")]
    public string Titulo { get; set; } = string.Empty;

    [JsonPropertyName("descripcion")]
    public string Descripcion { get; set; } = string.Empty;

    [JsonPropertyName("modalidad")]
    public string Modalidad { get; set; } = "remoto"; // "remoto" | "presencial"

    [JsonPropertyName("ubicacion")]
    public string? Ubicacion { get; set; }

    [JsonPropertyName("tipoCompensacion")]
    public string TipoCompensacion { get; set; } = "favor"; // "favor" | "dinero" | "ambos"

    [JsonPropertyName("cantidadDinero")]
    public float? CantidadDinero { get; set; }

    [JsonPropertyName("recompensaFavor")]
    public string? RecompensaFavor { get; set; }
}
